shader/main3d.vert.glsl

```diff
@@ -5,17 +5,27 @@ void main()
   vec3 norm = inNormal;
   vec3 normScreen = normalize(matNorm * norm);
 
+  cc_shade = inColor;
+  flags = inFlags;
+  geoMode = inGeoMode;
+  othermodeL = inOthermodeL;
+  othermodeH = inOthermodeH;
+
   vec4 lightTotal = vec4(ccData.ambientColor.rgb, 0.0);
   for(int i=0; i<2; ++i) {
     float lightStren = max(dot(norm, ccData.lightDir[i].xyz), 0.0);
     lightTotal += ccData.lightColor[i] * (lightStren * 2);
   }
-  
-  // Ambient light
-  cc_shade.rgb = linearToGamma(lightTotal.rgb);
-  cc_shade.a = 1.0;
-  cc_shade *= inColor;
+
+  lightTotal.rgb = linearToGamma(clamp(lightTotal.rgb, 0.0, 1.0));
+
+#if defined(F3DEX3)
+  cc_shade.rgb = geoModeSelect(G_LIGHTING, cc_shade.rgb, geoModeSelect(G_PACKED_NORMALS, lightTotal.rgb, cc_shade.rgb * lightTotal.rgb));
```

Comment:
Just to tidy it up you can move the nested geoModeSelect out into e.g.:
```glsl
vec3 shadeWithLight = geoModeSelect(G_PACKED_NORMALS, lightTotal.rgb, cc_shade.rgb * lightTotal.rgb);
cc_shade.rgb = geoModeSelect(G_LIGHTING, cc_shade.rgb, shadeWithLight);
```

As a side note:
I'm not sure if it can even happen in fast64, but you would be able to simulate wrong color-as-normals via:
`norm = geoModeSelect(SOME_FLAG, inColor * 2.0 - 1.0, inNormal)` at the very top

Reply:
it cannot happen with fast64 no, fast64 exports normals or vertex colors depending on g_lighting, it doesn´t let you export normals with no lighting because there is no scenario that I'm aware of that would benifit from it.
For example bowser in sm64, in my improved_models branch I simply just cleared the lighting geo mode using a seperate function before the displaylist, which is honestly cleaner than having a seperate geolayout option (sorry for the sm64 term but a geolayout is basically a list of commands to create displaylists)